### ISSUE TITLE
License info

### ISSUE THREAD

**Issue by TELSER1** [TheLegendAli/CCVL#2]
Hi,
I have a dataset that is weakly annotated (i.e., bounding boxes) and was looking into trying to utilize this code to run the algorithm described in http://arxiv.org/abs/1502.02734, and had a couple questions:
1.) As someone who can't feasibly modify the C++ code that's being wrapped/interfaced with, is repurposing this code for datasets other than PASCAL/VOC doable?  If so, are there any examples I could look at? 
2.)  What sort of license, if any, does this code fall under?

Thanks for any help, and for the python bindings to deeplab!

**Comment by sebastianffx** [comment]
Hello TELSER1,
I'm currently  pursuing (partially) your objective of using this code with another dataset (and for further improvements on the algorithm), so far I just want to reproduce the author's results, and up to now I'm able to clone the repo, recompile this branch and execute up to the line 59 of run_pascal.py, where an error about an image that does not exist happens, this error is because the missing annotations of the dataset of pascal voc augmented rewarding the section 4.1 of the paper that you cite, I have downloaded the extra annotations from the FAQ link: http://ccvl.stat.ucla.edu/deeplab_faq/ (Q:  A folder called /SegmentationClassAug/ does not exist in the original PASCAL VOC2012 Devkit) nevertheless the annotations there came in matlab format and I do not know how to export the required png's in such a way that they look the same as the ones provided in SegmentationClass of the original voc dataset. I presume that this is done adding both the boundaries and class segmentation matrices from the .mat of each image and then exporting this result as png (but I do not know the class - color convention used in voc or the threshold of the boundaries).
In summary: Looking deeper to the code, seems like its feasible to use them with another datasets, but you should understand well how first the CNN feats are created and then how to feed the CRF with those.
I hope to reproduce the results soon and use another datasets, then I'll tell you how it goes.

Cheers.

**Comment by TELSER1** [comment]
I emailed LC Chen about this; he says to look at deeplab-public / matlab / my_script / TransformBerkeleyVOC2011Annot.m  for converting the matlab files in Bharath's supplementary materials to the appropriate annotation format.  There are currently no available example prototxt files(I don't know about annotations) for the method using bounding box annotations in the project, although they are apparently to be released soon*.

**Comment by ghost** [comment]
When I apply the script, the original and changed image (SegmentationClassAug/...) look quite the same :(
